kgtk/cli/build_faiss.py

```diff
@@ -0,0 +1,111 @@
+"""
+Train and populate a faiss index that can compute nearest neighbors of given embeddings.
+"""
+
+from argparse import Namespace
+from kgtk.cli_argparse import KGTKArgumentParser
+
+
+def parser():
+    return {
+        'help': 'Train and populate a faiss index that can compute nearest neighbors of given embeddings.',
+        'description': 'Train and populate a faiss index that can compute nearest neighbors of given embeddings.'
+    }
+
+
+def add_arguments_extended(parser: KGTKArgumentParser, parsed_shared_args: Namespace):
+    """
+    Parse arguments
+    Args:
+        parser (kgtk.cli_argparse.KGTKArgumentParser)
+    """
+
+    # import modules locally
+    from kgtk.io.kgtkreader import KgtkReader, KgtkReaderOptions, KgtkReaderMode
+    from kgtk.utils.argparsehelpers import optional_bool    # not used yet
+    from kgtk.value.kgtkvalueoptions import KgtkValueOptions
+
+    _expert: bool = parsed_shared_args._expert
+
+    # REQUIRED #
+    # Related to input file
+    parser.add_argument('-i', '--input_file', '--embeddings_file', action='store', dest='embeddings_file',
+                        required=True, metavar="EMBEDDINGS_FILE",
+                        help='Input file containing the embeddings for which a Faiss index will be created.')
+
+    # Related to output
+    parser.add_argument('-o', '--output_file', '--index_file_out', action='store', dest='index_file_out',
+                        required=True, help="Output .idx file where the index fill be saved.",
+                        metavar="INDEX_FILE_OUT")
+
+    parser.add_argument('-id2n', '--index_to_node_file_out', action='store', dest='index_to_node_file_out',
```

Comment:
all the secondary options should only have `-`, instead of underscore.

`--index_to_node_file_out` should be `--index-to-node-file-out`.

Also as a general rule, do not use stop words like `to` in the parameter name, This review comment applies to all the subsequent parameters.

The `dest` parameter is fine.

Reply:
Is the reason to avoid stop words simply for brevity? I am struggling to think of a name for this that omits 'to' and isn't ambiguous. This parameter specifies the path where a kgtk file will be saved. The kgtk file contains a mapping of index to corresponding node. Do you have a suggestion for this parameter name?

Reply:
can you explain a bit more as to what exactly this file contains? mapping of what index ?how is this different from the output file?

Reply:
When you use a faiss index to search for nearest neighbors, it returns distances and corresponding indexes/IDs of the nearest neighbors, rather than the names or embeddings of the nearest neighbors. This file would allow you to look up the entity name that corresponds to the index/ID. Here's an example of what the file would look like:

Input embedding file:

`q1    <embedding>`
`q2    <embedding>`
`q3    <embedding>`

output index_to_node file:
`node1    label    node2`
`0   index_to_node    q1`
`1    index_to_node    q2`
`2    index_to_node    q3`

Reply:
I suppose I could use the word ID rather than index here to avoid confusion. Then I could call this 'node_id_file_out' to avoid 'to'.

Reply:
i see, using the stop word `to` makes sense here. I would call this `--faiss-id-to-node-mapping-file` As you can, brevity is not the issue, meaningful name is.

Reply:
done